pkg/plugins/shell/main_test.go

```diff
@@ -0,0 +1,53 @@
+package shell
+
+import "fmt"
+
+////////////// Test Utilities method aimed at mocking calls to other objects
+
+// mocking commandExecutor.ExecuteCommand
+type mockCommandExecutor struct {
+	gotCommand command
+	result     commandResult
+	err        error
+}
+
+func (mce *mockCommandExecutor) ExecuteCommand(cmd command) (commandResult, error) {
+	mce.gotCommand = cmd
+	if mce.gotCommand.Cmd == "" {
+		return commandResult{}, fmt.Errorf(ErrEmptyCommand)
+	}
```

Comment:
Is this necessary? 🤔

Reply:
It is not, in regard to your feedback about the "spec" checked before calling. If ExecuteCommand does NOT check for empty command (as it's the role of the caller), then this part should be deleted 👍

Reply:
Even if you don't: you could configure the mock in your case to raise an error instead of hard-coding this behavior.

Reply:
With the validation logic moved to shell.New, this code had been removed